### PR TITLE
Convert tests to opaque pointers

### PR DESCRIPTION
**Pull request by svenvh**
This completes the initial bulk conversion of tests using the migration script.

There are more tests to be converted; however they will need manual fixups, so leave them out of this bulk conversion.

### CODE REVIEW

File: test/layout.ll [hunk 1]
```diff
@@ -157,7 +157,7 @@ attributes #1 = { "less-precise-fpmad"="false" "no-frame-pointer-elim"="false" "
 !opencl.used.optional.core.features = !{!8}
 !opencl.compiler.options = !{!8}
 
-!0 = !{void (<3 x i32> addrspace(1)*)* @foo, !1, !2, !3, !4, !5}
+!0 = !{ptr @foo, !1, !2, !3, !4, !5}
```
Comment (vmaksimo) [inline test/layout.ll:160]: same

File: test/AtomicBuiltinsFloat.ll [hunk 2]
```diff
@@ -72,6 +72,6 @@ attributes #2 = { convergent nounwind }
 !1 = !{i32 2, i32 0}
 !2 = !{i32 3}
 !3 = !{!"none"}
-!4 = !{!"atomic_float*"}
+!4 = !{!"atomic_ptr"}
```
[the remark this answers is not in the document]
Comment (svenvh) [reply test/AtomicBuiltinsFloat.ll:75]: Good catch, seems the script was over-enthusiastic here.  I'll revert it.

File: test/simple.ll [hunk 3]
```diff
@@ -146,8 +143,8 @@ attributes #2 = { nounwind readnone }
 !4 = !{!"int*", !"int*", !"int"}
```
Comment (vmaksimo) [inline test/simple.ll:143]: I guess these should be changed as well as !8 and !9 below

**Review by vmaksimo — COMMENTED** [review]
One more finding :)

File: test/simple.ll (hunk 3 above)
Comment (svenvh) [reply test/simple.ll]: Good find! This belongs to the kernel arg metadata so shouldn't be changed. And neither should `!8` and `!9` so I've taken this file out now.

I also found a few more, which I've also taken out.

File: test/EnqueueEmptyKernel.ll [hunk 4]
```diff
@@ -36,38 +36,37 @@ target triple = "spir64-unknown-unknown"
...
-define internal spir_kernel void @__test_enqueue_empty_block_invoke_kernel(i8 addrspace(4)*) #3 {
+define internal spir_kernel void @__test_enqueue_empty_block_invoke_kernel(ptr addrspace(4)) #3 {
 entry:
-  call void @__test_enqueue_empty_block_invoke(i8 addrspace(4)* %0)
+  call void @__test_enqueue_empty_block_invoke(ptr addrspace(4) %0)
   ret void
 }
 
-declare i32 @__enqueue_kernel_basic_events(%opencl.queue_t*, i32, %struct.ndrange_t*, i32, %opencl.clk_event_t* addrspace(4)*, %opencl.clk_event_t* addrspace(4)*, i8 addrspace(4)*, i8 addrspace(4)*)
+declare i32 @__enqueue_kernel_basic_events(ptr, i32, ptr, i32, ptr addrspace(4), ptr addrspace(4), ptr addrspace(4), ptr addrspace(4))
```
Comment (MrSidims) [inline test/EnqueueEmptyKernel.ll:69]: Shouldn't it be target types? Guess script conversion doesn't handle such case.

Comment (svenvh) [reply test/EnqueueEmptyKernel.ll:69]: I suppose it doesn't matter here, as translation of `__enqueue_kernel_basic_events` into `OpEnqueueKernel` is known by the translator and it mainly needs to forward the pointer arguments.

**Review by vmaksimo — APPROVED** [review]

**Comment by svenvh** [comment]
> @svenvh if it's not a problem, could you please use separate commits for changes instead of force-push in future? :) It'd be much easier to review, mostly for such big changes

It may be a bit of a hidden GitHub feature, but you can see the diff between two force-pushes by clicking on the word "force-pushed" in "svenvh force-pushed the opaquify-tests branch ...".  But if incremental commits are preferred I can do that.

**Pull request merged**